Add 1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.py and 1342_Number_of_Steps_to_Reduce_a_Number_to_Zero.py

srivastavaayu opened the pull request:
Added Convert Binary Number in a Linked List to Integer (problem code - 1290) and Number of Steps to Reduce a Number to Zero (problem code - 1342)

srivastavaayu committed:
Create 1290_Convert_Binary_Number_In_A_Linked_List_To_Integer.py

Added 1290_Convert_Binary_Number_In_A_Linked_List_To_Integer.py (problem code - 1290)

srivastavaayu committed:
Create 1342_Number_of_Steps_to_Reduce_a_Number_to_Zero.py

Added 1342_Number_of_Steps_to_Reduce_a_Number_to_Zero.py (problem code - 1342)

srivastavaayu committed:
Update 1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.py

qiyuangong commented:
Hi @srivastavaayu 
Thank you for contribution! 👍 

Pls add one line description for your solution, e.g.,

```
| 12 | [Integer to Roman](https://leetcode.com/problems/integer-to-roman/) | [Python](https://github.com/qiyuangong/leetcode/blob/master/python/012_Integer_to_Roman.py) | [Background knowledge](http://www.rapidtables.com/convert/number/how-number-to-roman-numerals.htm) Just like 10-digit number, divide and minus |
```

qiyuangong commented on python/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.py:
```diff
@@ -0,0 +1,46 @@
+        ans=0
+        po=0
+        for i in range(len(ll)-1,-1,-1):
+            ans+=((2**po)*ll[i])
```
Add ll[i] >= 1 check will cut unnecessary computation. :)
```
            if ll[i]:
                ans+=((2**po)*ll[i])
```

qiyuangong commented on python/1342_Number_of_Steps_to_Reduce_a_Number_to_Zero.py:
```diff
@@ -0,0 +1,44 @@
+class Solution:
+    def numberOfSteps (self, num: int) -> int:
+        steps=0
+        while(num>0):
```
Nice solution. 

BTW, in https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/discuss/502809/just-count-number-of-0-and-1-in-binary, there is an interesting solution based on binary encoding.

qiyuangong commented:
Both solutions work well on leetcode.Good job.

Pls refine code style based on Pylint (you can install PEP8 or pylint plugin in you IDE), and add necessary comments to increase readability. :)

srivastavaayu committed:
Update 1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.py

Update 1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.py to include checks to cut unnecessary computation and change variable names to be more readable.

srivastavaayu committed:
Update README.md

Updated README.md to include files/solutions added.

srivastavaayu commented:
@qiyuangong I've made necessary changes as suggested by you. Please review the changes. And don't forget to change the links to the solutions in the README after pulling the commits.

qiyuangong commented:
> @qiyuangong I've made necessary changes as suggested by you. Please review the changes. And don't forget to change the links to the solutions in the README after pulling the commits.

OK. I will 👍

qiyuangong reviewed (approved):
LGTM
